server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -253,7 +261,11 @@ public boolean matchesField(String fieldName) {
         }
 
         @Override
-        public void process(IndexRequest indexRequest) {}
+        public void postProcess(IndexRequest indexRequest) {
+            if (indexRequest.id() == null && indexTracksRoutingId) {
+                indexRequest.id(TimeSeriesRoutingIdFieldMapper.encode(routingId));
```

Comment:
I think it is cleaner to add a routingHash field to `IndexRequest` instead of reusing the id property here.

Reply:
Went with using the routing param, seems to have no side effects.